src/sentry/sentry_apps/components.py

```diff
@@ -11,14 +11,15 @@
 from sentry.mediators.external_requests.select_requester import SelectRequester
 from sentry.models.integrations.sentry_app_component import SentryAppComponent
 from sentry.models.integrations.sentry_app_installation import SentryAppInstallation
+from sentry.services.hybrid_cloud.app.model import RpcSentryAppComponent, RpcSentryAppInstallation
 from sentry.services.hybrid_cloud.app.serial import serialize_sentry_app_installation
 from sentry.utils import json
 
 
 @dataclasses.dataclass
 class SentryAppComponentPreparer:
-    component: SentryAppComponent
-    install: SentryAppInstallation
+    component: SentryAppComponent | RpcSentryAppComponent
+    install: SentryAppInstallation | RpcSentryAppInstallation
```

Comment:
Would it make sense to make `install` an `RpcSentryAppInstallation` only, and add an init method to perform the serialization?

Reply:
That could work. I'd like to do that change separately though. There are still a few places where we pass in ORM objects that we could chase down.